app/src/main/java/com/realityexpander/tasky/presentation/agenda_screen/AgendaEvent.kt

```diff
@@ -0,0 +1,44 @@
+package com.realityexpander.tasky.presentation.agenda_screen
+
+import com.realityexpander.tasky.presentation.util.UiText
+
+sealed class AgendaEvent {
+    data class SetIsLoading(val isLoading: Boolean) : AgendaEvent()
+
+    data class UpdateUsername(val username: String) : AgendaEvent()
+    data class UpdateEmail(val email: String) : AgendaEvent()
+    data class UpdatePassword(val password: String) : AgendaEvent()
+    data class UpdateConfirmPassword(val confirmPassword: String) : AgendaEvent()
+    data class SetIsPasswordVisible(val isVisible: Boolean) : AgendaEvent()
+
+    object ValidateUsername : AgendaEvent()
+    object ValidateEmail : AgendaEvent()
+    object ValidatePassword : AgendaEvent()
+    object ValidateConfirmPassword : AgendaEvent()
+    object ValidatePasswordsMatch : AgendaEvent()
+
+    data class IsValidUsername(val isValid: Boolean) : AgendaEvent()
+    data class IsValidEmail(val isValid: Boolean) : AgendaEvent()
+    data class IsValidPassword(val isValid: Boolean) : AgendaEvent()
+    data class IsValidConfirmPassword(val isValid: Boolean) : AgendaEvent()
+    data class IsPasswordsMatch(val isMatch: Boolean) : AgendaEvent()
```

Comment:
These classes all sound like something you'd get a result from(the resulting boolean). But in the end, such events just do something or trigger some kind of action, so naming-wise I'd prefer something like `ValidateUsername` or `CheckIfPasswordsMatch` here to make it clear that the UI can trigger such functionality.

Reply:
OK, I will implement this suggestion. I see I was over complicating things here.